types/src/message.rs

```diff
@@ -17,62 +18,89 @@ use serde::{Deserialize, Serialize};
 /// Incoming message
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 #[serde(bound(deserialize = ""))]
-pub struct Message<TYPES: NodeTypes> {
+pub struct Message<
+    TYPES: NodeType,
+    LEAF: LeafType<NodeType = TYPES>,
+    PROPOSAL: ProposalType<NodeType = TYPES>,
+> {
     /// The sender of this message
     pub sender: TYPES::SignatureKey,
 
     /// The message kind
-    pub kind: MessageKind<TYPES>,
+    pub kind: MessageKind<TYPES, LEAF, PROPOSAL>,
```

Comment:
Just a note to remember that this, in particular, is going to need to be refactored back out in the work we do next year. We want to have the MessageKind more customized to the consensus layer, which probably means taking the specific messages out of the struct, and entirely redoing this enum.

Reply:
Sounds good! I just added a TODO comment to the enum.